packages/netlify-cms-core/src/actions/editorialWorkflow.ts

```diff
@@ -406,7 +414,10 @@ export function persistUnpublishedEntry(collection: Collection, existingUnpublis
         }),
       );
       dispatch(unpublishedEntryPersisted(collection, transactionID, newSlug));
-      if (!existingUnpublishedEntry) return dispatch(loadUnpublishedEntry(collection, newSlug));
+      if (entry.get('slug') !== newSlug) {
+        dispatch(loadUnpublishedEntry(collection, newSlug));
+        navigateToEntry(collection.get('name'), newSlug);
```

Comment:
`navigateToEntry` was done in `componentDidUpdate` in the editor and moved here.